src/fde.h

```diff
@@ -84,8 +84,25 @@ class fde
     /// use I/O methods that maintain an internal-to-them buffer
     void useBufferedIo(READ_HANDLER *, WRITE_HANDLER *);
 
-    int read(int fd, char *buf, int len) { return readMethod_(fd, buf, len); }
-    int write(int fd, const char *buf, int len) { return writeMethod_(fd, buf, len); }
+    /// call the read method set by setIo()
+    /// Performs I/O counting for all bytes received.
+    /// \returns the read method result
+    int read(int fd, char *buf, int len) {
+        auto n = readMethod_(fd, buf, len);
+        if (n > 0)
+            bytesRead(n);
+        return n;
+    }
+
+    /// call the write method set by setIo()
```

Comment:
```suggestion
    /// Call the write method set by setIo().
```
or, better,

```suggestion
    /// Call the write method set by setIo() or similar.
```

or simply

```suggestion
    /// Call the previously set write I/O method.
```